bentoml/yatai/__init__.py

```diff
@@ -75,16 +76,34 @@ def ApplyDeployment(self, request, context=None):
                 request.deployment.namespace or self.default_namespace
             )
 
-            # TODO: validate deployment config
+            previous_deployment = self.deployment_store.get(
+                request.deployment.name, request.deployment.namespace
+            )
+            if previous_deployment:
+                # check deployment platform
+                if previous_deployment.spec.operator != request.deployment.spec.operator:
+                    return ApplyDeploymentResponse(
+                        status=Status.ABORTED(
+                            'New deployment spec has different platform from existing one. '
+                            'Please delete existing deployment and apply again'
+                        )
+                    )
+                request.deployment.state = previous_deployment.state
 
-            # create or update deployment spec record
-            self.deployment_store.insert_or_update(request.deployment)
+                request.deployment.state.state = DeploymentState.PENDING
+                if not request.deployment.labels:
```

Comment:
l94-l97 are probably not necessary - how can a user remove all labels if they want to?

Reply:
Right now, there isn't way to remove labels yet.  Maybe that's something web could provide as a feature?

I am thinking persevering the old labels might be more important than removing extra labels for now, that's why i add it.

The scenario I keep come back to, is when they update the deployment with new bento tag (maybe the most common tasks compare to others).  I don't want the labels get wiped out, or I don't want they have to re-enter labels everytime, when they apply new bento tag.